Add link to open fork settings when creating new branch on a forked repository

geophilusd opened the pull request:
Closes #12534 

## Description
<!--
A summary of the changes made along with any other information that would be helpful to a reviewer such as potential tradeoffs or alternative approaches you considered.
-->
- Added link to open fork settings when creating a new branch on a forked repository.

### Screenshots

<!--
If this PR touches the UI layer of the app, please include screenshots or animated gifs to show the changes.
-->
<img width="455" alt="Screenshot 2022-02-18 at 12 50 05 PM" src="https://user-images.githubusercontent.com/35450302/154636291-63d41cc2-e62c-4091-9e90-aab66d83bfc5.png">
<img width="455" alt="Screenshot 2022-02-18 at 12 30 21 PM" src="https://user-images.githubusercontent.com/35450302/154635995-5f721605-b06b-4e1a-9515-65e5ab46008d.png">


## Release notes

<!--
You can leave this blank if you're not sure.
If you don't believe this PR needs to be mentioned in the release notes, write "Notes: no-notes".
-->

Notes:
Added link to open fork settings when creating a new branch on a forked repository.

Commit:
Add link to open fork settings when creating new branch on forked repo

Commit:
Merge remote-tracking branch 'upstream/development' into fork_message

Commit:
Add link in both states of forked repo

haykam821 commented on app/src/ui/create-branch/create-branch-dialog.tsx:
```diff
@@ -422,8 +430,28 @@ export class CreateBranch extends React.Component<
+    if (isRepositoryWithForkedGitHubRepository(this.props.repository)) {
+      return (
+        <p>
+          Fork behaviour can be changed from {' '}
```
This should probably match the American English spelling used in repository settings:

```suggestion
          Fork behavior can be changed from {' '}
```

Commit:
Spelling and lint

villabellissima reviewed (commented):
Hello😶

tidy-dev reviewed (changes requested):
@geophilusd Thank you for wanting to improve the fork new branch interaction! 

I have a few small requests.

In other cases where we want to inform the user of a reason, but it may not be something they actually use regularly, we put the note in less eye catching text. 

As done in the commit message warning, "You can also choose..."
![image](https://user-images.githubusercontent.com/75402236/152565239-30fd92f8-03a6-4843-ab71-3735409113c0.png)

Here is a link to the code for that, 
https://github.com/desktop/desktop/blob/17afca94bb98a4edca6b21ccf539074945553133/app/src/ui/changes/commit-message-avatar.tsx#L154

Other thoughts are that "Fork behavior can be changed" does indicate that something on this dialog is impacted, but doesn't really tell us what is being impacted by it. What do you think of changing the verbiage "Your default branch source is determined by your `<link>` fork behavior settings`</link>`."

Also, usually we place these sort of notes directly underneath the input they impact. In this case, I think it should be placed underneath the branch selection. 

Similarly, I think this kind of note needs to be placed in the dialog when there is not a branch selection as well. (see first if block of the `renderForkBranchSelection` method)

Commit:
Merge remote-tracking branch 'upstream/development' into fork_message

Commit:
Move text below input, use secondary color

Commit:
Add fork link when creating new branch from default branch

Commit:
minor text change

Commit:
lint

geophilusd commented:
@tidy-dev I have made the suggested changes. Updated the screenshots in the PR description. Please check and let me know if I need to change something.

tidy-dev commented on app/src/ui/create-branch/create-branch-dialog.tsx:
```diff
@@ -348,12 +351,13 @@ export class CreateBranch extends React.Component<
           ). <Ref>{currentBranchName}</Ref> is the {defaultBranchLink} for your
           repository.
-        </p>
+          {this.renderForkLink()}
```
I think we can be a little more concise in these "non-selection" instances by conditionally adding prefix to the existing informative text of: ```
As determined by your `<link>` for behavior settings `</link>`, your new branch will be based on your currently checked out branch. ```  Just got to be mindful of the 'y' vs 'Y' depending on if it is a fork or not.

geophilusd replied:
I have changed it to a suffix instead. Does that work?

tidy-dev commented on app/src/ui/create-branch/create-branch-dialog.tsx:
```diff
@@ -395,12 +404,13 @@ export class CreateBranch extends React.Component<
           's {defaultBranchLink} (
           <Ref>{upstreamDefaultBranch.nameWithoutRemote}</Ref>).
-        </p>
+          {this.renderForkLink()}
```
Same as comment on `renderRegulartBranchSelection` - Use prefix instead of additional sentence.. probably a good place for another method like `renderForkLinkPrefix`

tidy-dev commented on app/src/ui/create-branch/create-branch-dialog.tsx:
```diff
@@ -422,8 +432,28 @@ export class CreateBranch extends React.Component<
+    if (isRepositoryWithForkedGitHubRepository(this.props.repository)) {
+      return (
+        <div className="secondary-text">
+          Your default branch source will be determined by your{' '}
```
```suggestion
          Your default branch source is determined by your{' '}
```

geophilusd replied:
changed

tidy-dev commented on app/src/ui/create-branch/create-branch-dialog.tsx:
```diff
@@ -422,8 +432,28 @@ export class CreateBranch extends React.Component<
+      return (
+        <div className="secondary-text">
+          Your default branch source will be determined by your{' '}
+          <LinkButton onClick={this.onForkSettingsClick}>
```
This link is not displaying inline and seems to wrap oddly. Think we may want to add a css rule to the `.link-button-component` to be `display: inline;` when used in this dialog.

geophilusd replied:
added 👍🏻 
Now the wrapping seems to be fine.

tidy-dev reviewed (changes requested):
@geophilusd Thank you for those changes! The secondary text under the options definitely feels more consistent to the rest of app.

However, for the instances where there are not branch options, I think it ends up being a little too verbose. I have commented a few suggestions to make it more concise and hopefully easier to digest at a glance.

Commit:
Merge remote-tracking branch 'upstream/development' into fork_message

Commit:
review changes

shorter message for branch creation from default branch
minor text change
link wrapping fix

Commit:
code formatting

geophilusd commented:
@tidy-dev I have pushed some changes based on your suggestions. Please have a look.

tidy-dev reviewed (changes requested):
@geophilusd This is so close! 

However, one state that feels confusing is when we see (fork settings is non-contributing)
![image](https://user-images.githubusercontent.com/75402236/156931417-34687fda-93be-4f62-b4e2-68422f6045d5.png)
It seems to indicate that development is the default branch because of the fork settings, but that is not the case. This is just an informational sentence about default branches.

It is the first sentence that is influenced by the setting, we need: 
```
Your new branch will be based on your currently checked out branch (development) as determined by your fork settings. development is your default branch.
```

Commit:
format text based on suggestion

geophilusd commented:
@tidy-dev done

tidy-dev reviewed (approved):
✨  Thank you @geophilusd for making forking a little easier for new fork contributors in desktop!
